server/src/main/java/org/elasticsearch/search/aggregations/bucket/significant/SignificantTermsAggregationBuilder.java

```diff
@@ -149,6 +163,11 @@ protected AggregationBuilder shallowCopy(Builder factoriesBuilder, Map<String, O
     protected void innerWriteTo(StreamOutput out) throws IOException {
         bucketCountThresholds.writeTo(out);
         out.writeOptionalString(executionHint);
+        if (out.getVersion().onOrAfter(Version.V_7_0_0_alpha1)) {
+            out.writeOptionalWriteable(collectMode);
+        } else {
```

Comment:
A couple comments:
- Because I've taken so long to review, we'll need to update this version constant to be the latest (`V_8_0_0`). Once it's backported to previous versions, I'll make sure to adjust it.
- If the node we're writing to is older than the current version, then writing a collection mode will result in an error, since that node is not expecting a field here. I'm going to check with the rest of the team what would be best to do in this case, and will get back to you. We could either skip the collection mode, or throw an error here.

Reply:
I don't think we should have this option in the builder but if we do I don't think we should throw an error either. This option should be treated as best effort so IMO it is ok to just ignore it for older nodes.